paasta_tools/cli/cmds/remote_run.py

```diff
@@ -0,0 +1,314 @@
+#!/usr/bin/env python
+# Copyright 2015-2017 Yelp Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import sys
+import os
+import shlex
+import service_configuration_lib
+import re
+
+sys.path.insert(0, '/usr/lib/python2.7/site-packages/mesos')
+from mesos.interface import Scheduler
+from mesos.interface import mesos_pb2
+
+from paasta_tools.cli.utils import figure_out_service_name
+from paasta_tools.cli.utils import lazy_choices_completer
+from paasta_tools.cli.utils import list_instances
+from paasta_tools.cli.utils import list_clusters
+from paasta_tools.cli.utils import list_services
+from paasta_tools.cli.utils import run_on_master
+from paasta_tools.utils import load_system_paasta_config
+from paasta_tools.utils import paasta_print
+from paasta_tools.utils import PaastaColors
+from paasta_tools.utils import PaastaNotConfiguredError
+from paasta_tools.utils import SystemPaastaConfig
+from paasta_tools.utils import DEFAULT_SOA_DIR
+from paasta_tools.utils import validate_service_instance
+
+from paasta_tools.native_mesos_scheduler import create_driver_with
+from paasta_tools.native_mesos_scheduler import compose_job_id
+from paasta_tools.native_mesos_scheduler import load_paasta_native_job_config
+
+
+class PaastaAdhocScheduler(Scheduler):
```

Comment:
I wonder whether it would be better to have this inherit from PaastaNativeScheduler. I think you'd basically just have to override `need_more_tasks` to return True until you start a task, and maybe `kill_tasks_if_necessary` to do nothing.